src/basic/env-util.c

```diff
@@ -137,7 +137,7 @@ bool strv_env_name_or_assignment_is_valid(char **l) {
         return true;
 }
 
-static int env_append(char **e, char ***k, char **a) {
+static int env_append(char **e, char ***k, char ** restrict a) {
```

Comment:
for static, local functions this is also useless, compiler has everything available to figure this out on its own, we don't need to help it. thi is really just noise I'd say